ts/features/walletV3/payment/screens/WalletPaymentPickMethodScreen.tsx

```diff
@@ -95,24 +104,61 @@ const WalletPaymentPickMethodScreen = () => {
   // only when the transaction status becomes ACTIVATED.
   useOnTransactionActivationEffect(
     React.useCallback(() => {
+      pipe(
+        sequenceT(O.Monad)(
+          pot.toOption(paymentAmountPot),
+          pot.toOption(transactionPot),
+          selectedWalletOption
+        ),
+        O.map(([paymentAmountInCents, transaction, selectedWallet]) => {
+          const transferList = transaction.payments.reduce(
+            (a, p) => [...a, ...(p.transferList ?? [])],
+            [] as ReadonlyArray<Transfer>
+          );
+          const paymentToken = transaction.payments[0]?.paymentToken;
+
+          dispatch(
+            walletPaymentCalculateFees.request({
+              paymentToken,
+              paymentMethodId: selectedWallet.paymentMethodId,
+              walletId: selectedWallet.walletId,
+              paymentAmount: paymentAmountInCents,
+              transferList
+            })
+          );
+        })
+      );
+      setWaitingTransactionActivation(false);
+    }, [dispatch, paymentAmountPot, selectedWalletOption, transactionPot])
+  );
+
+  React.useEffect(() => {
+    const pspList = pot.getOrElse(pspListPot, []);
+    if (pspList.length > 1) {
```

Comment:
What happens if psp list is empty?

Reply:
There is another action that is called at the screen focus inside a `useFocusEffect`, more specifically the `walletPaymentResetPspList` that resets the PSP List when the user lands on the `WalletPaymentPickMethodScreen`.

Anyway if the `pspList` is empty, should be shown an error since without any PSP the payment cannot be processed.

What do you think about the implementation in the commit f3500ca1fb575010d26f5d37a622b02bedae3034?